hummingbot/market/kraken/kraken_api_order_book_data_source.py

```diff
@@ -0,0 +1,278 @@
+#!/usr/bin/env python
+
+import aiohttp
+import asyncio
+import gzip
+import json
+import logging
+import pandas as pd
+import time
+from typing import (
+    Any,
+    AsyncIterable,
+    Dict,
+    List,
+    Optional,
+)
+import ujson
+import websockets
+from websockets.exceptions import ConnectionClosed
+
+from hummingbot.core.data_type.order_book import OrderBook
+from hummingbot.core.data_type.order_book_message import OrderBookMessage
+from hummingbot.core.data_type.order_book_tracker_data_source import OrderBookTrackerDataSource
+from hummingbot.core.data_type.order_book_tracker_entry import OrderBookTrackerEntry
+from hummingbot.core.utils import async_ttl_cache
+from hummingbot.core.utils.async_utils import safe_gather
+from hummingbot.logger import HummingbotLogger
+from hummingbot.market.kraken.kraken_order_book import KrakenOrderBook
+
+KRAKEN_SYMBOLS_URL = "https://api.kraken.com/0/public/Assets"
+KRAKEN_MARKETS_URL = "https://api.kraken.com/0/public/AssetPairs"
+KRAKEN_TICKER_URL = "https://api.kraken.com/0/public/Ticker"
+KRAKEN_DEPTH_URL = "https://api.kraken.com/0/public/Depth"
+KRAKEN_WS_URI = "wss://ws.kraken.com"
+
+
+class KrakenAPIOrderBookDataSource(OrderBookTrackerDataSource):
+    MESSAGE_TIMEOUT = 30.0
+    PING_TIMEOUT = 10.0
+
+    _haobds_logger: Optional[HummingbotLogger] = None
+
+    @classmethod
+    def logger(cls) -> HummingbotLogger:
+        if cls._haobds_logger is None:
+            cls._haobds_logger = logging.getLogger(__name__)
+        return cls._haobds_logger
+
+    def __init__(self, symbols: Optional[List[str]] = None):
+        super().__init__()
+        self._symbols: Optional[List[str]] = symbols
+
+    @classmethod
+    @async_ttl_cache(ttl=60 * 30, maxsize=1)
+    async def get_active_exchange_markets(self) -> pd.DataFrame:
+        async with aiohttp.ClientSession() as client:
+            exchange_response = await client.get(KRAKEN_MARKETS_URL)
+            exchange_response: aiohttp.ClientResponse = exchange_response
+            if exchange_response.status != 200:
+                raise IOError(f"Error fetching Kraken exchange information. "
+                              f"HTTP status is {exchange_response.status}.")
+            exchange_data = await exchange_response.json()
+            trading_pairs = {
+                trading_pair: {
+                    "symbol": f"{info['base']}/{info['quote']}",
```

Comment:
Kraken's trading pair convention resulting in `KeyError` issue with `symbol`
In Kraken:
BTC->XXBT, ETH->ZETH, USD->ZUSD